src/Rewards/StakingContractMainnet.sol

```diff
@@ -74,11 +75,14 @@ contract StakingContractMainnet is ReentrancyGuard {
     event Unsubscribe(uint256 indexed id, address indexed user);
     event Claim(uint256 indexed id, address indexed user, uint256 amount);
 
-    function createIncentive(address token, address rewardToken, uint112 rewardAmount, uint32 startTime, uint32 endTime)
-        external
-        nonReentrant
-        returns (uint256 incentiveId)
-    {
+    function createIncentive(
+        address token,
+        address rewardToken,
+        uint112 rewardAmount,
+        uint32 startTime,
+        uint32 endTime,
+        bool isRewardRounded
+    ) external nonReentrant returns (uint256 incentiveId) {
```

Comment:
I'd prefer verifying if the **token** is a VaultToken when creating the incentive, when `isRewardRounded` is set to true. Also might want to disallow rounding for regular ERC20s

Reply:
As discussed, we'll keep it as-is and assume all incentive streams we create will be rounded only if they're vaults